src/AFQMC/HamiltonianOperations/THCOps.hpp

```diff
@@ -133,43 +133,43 @@ class THCOps
         vn0(std::move(v0_)),
         E0(e0_)
   {
-    gnmu    = Luv.size(1);
-    grotnmu = rotMuv.size(1);
+    gnmu    = std::get<1>(Luv.sizes());
```

Comment:
well its ugly, but I imagine you had your reasons to drop the `size_t size(int)` interface.
I assume there is no performance impact or an improvement.  I guess you also get compile time bounds checking.

Reply:
yes, this is a current defect in C++, it is ugly to access aggregates by index, but the code advantages are there: 
The compile-time access to the index allows the compiler to generate the internal sizes as necessary.
Also, compile-time access allows to check for bound errors (in the number of dimensions) at compile-time.